src/lib.rs

```diff
@@ -8,35 +8,270 @@ mod transaction;
 pub mod update;
 
 use crate::{
-    keys::{OwnershipKeyPair, OwnershipPublicKey, PublishingKeyPair, RevocationKeyPair},
+    create::{BuildFundingPSBT, SignFundingPSBT},
+    keys::{
+        OwnershipKeyPair, OwnershipPublicKey, PublishingKeyPair, PublishingPublicKey,
+        RevocationKeyPair, RevocationPublicKey, RevocationSecretKey,
+    },
     transaction::{CommitTransaction, FundingTransaction, SplitTransaction},
 };
 use anyhow::bail;
-use bitcoin::Amount;
+use bitcoin::{Amount, Transaction};
 use ecdsa_fun::adaptor::EncryptedSignature;
-use keys::{PublishingPublicKey, RevocationPublicKey, RevocationSecretKey};
+
+#[allow(dead_code)]
+pub struct Channel {
+    x_self: OwnershipKeyPair,
+    X_other: OwnershipPublicKey,
+    pub TX_f_body: FundingTransaction,
+    pub current_state: ChannelState,
+    revoked_states: Vec<RevokedState>,
+}
+
+#[async_trait::async_trait]
+pub trait BroadcastSignedTransaction {
+    async fn broadcast_signed_transaction(&self, transaction: Transaction) -> anyhow::Result<()>;
+}
+
+#[async_trait::async_trait]
+pub trait SendMessage {
+    async fn send_message(&mut self, message: create::Message) -> anyhow::Result<()>;
+}
+
+#[async_trait::async_trait]
+pub trait ReceiveMessage {
+    async fn receive_message(&mut self) -> anyhow::Result<create::Message>;
+}
+
+impl Channel {
+    pub async fn create_alice<N, W>(
+        network: &mut N,
+        wallet: &W,
+        // Both parties fund the same amount
+        fund_amount: Amount,
+        time_lock: u32,
+    ) -> anyhow::Result<Self>
+    where
+        W: BuildFundingPSBT + SignFundingPSBT + BroadcastSignedTransaction,
+        N: SendMessage + ReceiveMessage,
+    {
+        let alice0 = create::Alice0::new(fund_amount, time_lock);
+
+        let message0_alice = alice0.next_message();
+        network
+            .send_message(create::Message::Message0(message0_alice))
+            .await?;
+
+        let message0_bob: create::Message0 = match network.receive_message().await? {
+            create::Message::Message0(message) => message,
+            _ => anyhow::bail!("wrong message"),
+        };
```

Comment:
```suggestion
        let message0_bob: create::Message0 = if let create::Message::Message0(message) = network.receive_message().await? {
             message    
        } else {
        	_ => anyhow::bail!("wrong message"),
        };
```
Maybe more readable? nitpick.